src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -319,7 +392,9 @@ impl<'a, 'tcx> Qualifier<'a, 'tcx, 'tcx> {
                         // Check for unused values. This usually means
                         // there are extra statements in the AST.
                         for temp in mir.temps_iter() {
-                            if self.local_qualif[temp].is_none() {
+                            if !self.mut_interior_checker.local_mut_interior.contains(&temp) &&
+                               !self.drop_checker.local_needs_drop.contains(&temp) &&
+                               !self.local_not_const.contains(&temp) {
```

Comment:
This replacement is wrong. What `None` meant was "uninitialized", whereas this is checking for "no flags set" aka "promotable".